spring-integration-redis/src/test/java/org/springframework/integration/redis/inbound/ReactiveRedisStreamMessageProducerTests.java

```diff
@@ -162,6 +170,34 @@ public void testReadingMessageAsConsumerInConsumerGroup() {
 				.verify(Duration.ofSeconds(10));
 	}
 
+	@Test
+	@RedisAvailable
+	public void testReadingPendingMessageWithNoAutoACK() {
+		Address address = new Address("Winterfell, Westeros");
+		Person person = new Person(address, "John Snow");
+
+		this.template.opsForStream()
+					 .createGroup(STREAM_KEY, this.redisStreamMessageProducer.getBeanName())
+					 .as(StepVerifier::create)
+					 .assertNext(message -> assertThat(message).isEqualTo("OK"))
+					 .thenCancel()
+					 .verify(Duration.ofSeconds(10));
+
+		this.redisStreamMessageProducer.setCreateConsumerGroup(false);
+		this.redisStreamMessageProducer.setAutoAck(false);
+		this.redisStreamMessageProducer.setConsumerName(CONSUMER);
+		this.redisStreamMessageProducer.afterPropertiesSet();
+		this.redisStreamMessageProducer.start();
```

Comment:
Any chances to see consumed messages and verify their headers for that `IntegrationMessageHeaderAccessor.ACKNOWLEDGMENT_CALLBACK`?


And probably after `pending` verification call the ack and see if there is no pending afterward left.